nimbess-cni.go

```diff
@@ -0,0 +1,314 @@
+// Copyright (c) 2019 Red Hat and/or its affiliates.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Main package for the Nimbess CNI plugin
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net"
+	"os"
+	"time"
+
+	"github.com/containernetworking/cni/pkg/skel"
+	"github.com/containernetworking/cni/pkg/types"
+	"github.com/containernetworking/cni/pkg/version"
+	"google.golang.org/grpc"
+
+	cnitypes "github.com/containernetworking/cni/pkg/types/current"
+	bv "github.com/containernetworking/plugins/pkg/utils/buildversion"
+	cninimbess "github.com/nimbess/nimbess-agent/pkg/proto/cni"
+	log "github.com/sirupsen/logrus"
+)
+
+const defaultLogFile = "/tmp/nimbess-cni.log"
```

Comment:
should this be /var/log/nimbess/nimbess-cni.log?

Reply:
+1

Reply:
is /var/log always right? Just wondering if centos/fedora is different than ubuntu for instance and then wondering if there is some env var that sets it accordingly. No need to change the code here beyond using var/log since we are just fedora with no ubuntu plan anyways, so this is more of a note.

Reply:
/var/log is good we dont plan to support other platforms right now